SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1824,3 +1827,103 @@ void USpatialNetDriver::DelayedSendDeleteEntityRequest(Worker_EntityId EntityId,
 		Sender->SendDeleteEntityRequest(EntityId);
 	}, Delay, false);
 }
+
+void USpatialNetDriver::HandleStartupOpQueueing(const TArray<Worker_OpList*>& InOpLists)
+{
+	if (InOpLists.Num() == 0)
+	{
+		return;
+	}
+
+	QueuedStartupOpLists.Append(InOpLists);
+	bIsReadyToStart = FindAndDispatchStartupOps(InOpLists);
+
+	if (!bIsReadyToStart)
+	{
+	    return;
+	}
+
+	// We've found and dispatched all ops we need for startup, trigger BeginPlay()
+	// on the GSM and process the queued ops.  Note that FindAndDispatchStartupOps()
+	// will have notified the Dispatcher to skip the startup ops that we've
+	// processed already.
+	GlobalStateManager->TriggerBeginPlay();
+
+	for (Worker_OpList* OpList : QueuedStartupOpLists)
+	{
+		Dispatcher->ProcessOps(OpList);
+		Worker_OpList_Destroy(OpList);
+	}
+
+	// Sanity check that the dispatcher encountered, skipped, and removed
+	// all Ops we asked it to skip
+	check(Dispatcher->GetNumOpsToSkip() == 0);
+
+	QueuedStartupOpLists.Empty();
+}
+
+bool USpatialNetDriver::FindAndDispatchStartupOps(const TArray<Worker_OpList*>& InOpLists)
+{
+	TArray<Worker_Op*> FoundOps;
+
+	// Search for entity id reservation response and process it.  The entity id reservation
+	// can fail to reserve entity ids.  In that case, the EntityPool will not be marked ready,
+	// a new query will be sent, and we will process the new response here when it arrives.
+	if (!EntityPool->IsReady())
+	{
+		Worker_Op* EntityIdReservationResponseOp = nullptr;
+		FindFirstOpOfType(InOpLists, WORKER_OP_TYPE_RESERVE_ENTITY_IDS_RESPONSE, &EntityIdReservationResponseOp);
+
+		if (EntityIdReservationResponseOp != nullptr)
+		{
+			FoundOps.Add(EntityIdReservationResponseOp);
+		}
+	}
+
+	// Search for StartupActorManager ops we need and process them
+	if (!GlobalStateManager->IsReadyToCallBeginPlay())
+	{
+		Worker_Op* AddComponentOp = nullptr;
+		FindFirstOpOfTypeForComponent(InOpLists, WORKER_OP_TYPE_ADD_COMPONENT, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID, &AddComponentOp);
+
+		Worker_Op* AuthorityChangedOp = nullptr;
+		FindFirstOpOfTypeForComponent(InOpLists, WORKER_OP_TYPE_AUTHORITY_CHANGE, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID, &AuthorityChangedOp);
+
+		// If we are going to get both ops, we expect them in the same Worker_OpList
+		check(AddComponentOp != nullptr || (AuthorityChangedOp == nullptr));
```

Comment:
I'm a bit confused about this `check`. Could you give a bit more detail as to what it's checking for?

Reply:
This check is basically "if we didn't receive the AddComponent Op, then make certain we didn't receive the AuthorityChangedOp".  It verifying my assumption that these ops must be received together for the auth server, which I wasn't 100% sure of.  The assert has not fired in all my testing so far.